image-swipe.android.ts

```diff
@@ -471,18 +471,41 @@ class ZoomImageView extends android.widget.ImageView {
 
 class OrientationListener extends android.view.OrientationEventListener {
     private _zoomImageView: WeakRef<ZoomImageView>;
+    private _previousOrientation: number;
 
     constructor(context: android.content.Context, zoomImageView: WeakRef<ZoomImageView>) {
         super(context);
 
         this._zoomImageView = zoomImageView;
+        this._previousOrientation = ORIENTATION_PORTRAIT;
 
         return __native(this);
     }
 
     public onOrientationChanged(orientation: number) {
         const zoomImageView: ZoomImageView = this._zoomImageView.get();
-        if (zoomImageView) {
+        var orientationChanged = false;
+        var currentOrientation;
+
+        if (orientation <= 45) {
+            currentOrientation = ORIENTATION_PORTRAIT;
+        } else if (orientation <= 135) {
+            currentOrientation = ORIENTATION_LANDSCAPE_REVERSE;
+        } else if (orientation <= 225) {
+            currentOrientation = ORIENTATION_PORTRAIT_REVERSE;
+        } else if (orientation <= 315) {
+            currentOrientation = ORIENTATION_LANDSCAPE;
+        } else {
+            currentOrientation = ORIENTATION_PORTRAIT;
+        }
+        
+        if (currentOrientation != this._previousOrientation) {
+            this._previousOrientation = currentOrientation;
+            orientationChanged = true;
+        }
+        
+        if (zoomImageView && orientationChanged) {
+            // ToDo: Rotate the image based on currentOrientation?
```

Comment:
Not sure what do you mean with this TODO. Currently the image rotates as the phone rotates to the next orientation.